extension-workspace-reference.md

```diff
@@ -0,0 +1,82 @@
+# workspace/reference extension to LSP
+
+The `workspace/reference` extension to the Language Server Protocol (LSP) enables a language server to export all of the references to dependencies that code inside of the workspace makes.
+
+Use case: clients of a language server can invoke `workspace/reference` in order to find external references to dependencies. This information can then be stored inside of a database, which allows the caller to create a 'global mapping' of symbols in dependencies to the workspace they are used in (e.g. to see "how do other people use this symbol?").
+
+### Initialization
+
+`ServerCapabilities` may contain a new field to indicate server-side support for this extension:
+
+```typescript
+interface ServerCapabilities {
+  /* ... all fields from the base ServerCapabilities ... */
+
+  /**
+   * The server provides workspace reference exporting support.
+   */
+  workspaceReferenceProvider?: boolean;
```

Comment:
Hmm, okay, I see your point. I don't think those translate quite as nicely to Go like that, but to be honest it probably doesn't matter that much. I'll stop bikeshedding :) It sounds like I should use `xworkspaceReferencesProvider` as the capability and `workspace/xreference` (NOT with `s` suffix like you said in your reply here, just to be clear)